lib/Doctrine/Migrations/SchemaDumper.php

```diff
@@ -166,27 +149,14 @@ private static function pregMatch(string $pattern, string $subject, ?array &$mat
         }
 
         if ($ret === false) {
-            switch (preg_last_error()) {
-                case PREG_INTERNAL_ERROR:
-                    $error = sprintf('Internal PCRE error, please check your Regex. Reported errors: %s.', implode(', ', $errorMessages));
-                    break;
-                case PREG_BACKTRACK_LIMIT_ERROR:
-                    $error = 'pcre.backtrack_limit reached.';
-                    break;
-                case PREG_RECURSION_LIMIT_ERROR:
-                    $error = 'pcre.recursion_limit reached.';
-                    break;
-                case PREG_BAD_UTF8_ERROR:
-                    $error = 'Malformed UTF-8 data.';
-                    break;
-                case PREG_BAD_UTF8_OFFSET_ERROR:
-                    $error = 'Offset doesn\'t correspond to the begin of a valid UTF-8 code point.';
-                    break;
-                default:
-                    $error = 'Error.';
-            }
-
-            throw new InvalidArgumentException($error);
+            throw new InvalidArgumentException(match (preg_last_error()) {
+                PREG_INTERNAL_ERROR => sprintf('Internal PCRE error, please check your Regex. Reported errors: %s.', implode(', ', $errorMessages)),
+                PREG_BACKTRACK_LIMIT_ERROR => 'pcre.backtrack_limit reached.',
```

Comment:
this could be replaced by using `preg_last_error_message` instead

Reply:
I guess.